reframe/core/pipeline.py

```diff
@@ -197,11 +197,15 @@ def disable_hook(self, hook_name):
 
     @classmethod
     def pipeline_hooks(cls):
+        # The hook registry stores hooks in MRO order, but we want to attach
+        # the hooks in reverse order so that the more specialized hooks (those
+        # at the beginning of the MRO list) will be executed last
+
         ret = {}
         for hook in cls._rfm_hook_registry:
             for stage in hook.stages:
                 try:
-                    ret[stage].append(hook.fn)
+                    ret[stage].insert(0, hook.fn)
```

Comment:
Do you think that you can use a `deque` from `collections` and the `appendleft` method?

Reply:
Good point, the difference is humongous in performance (that's the numbers for 100K inserts):

```
Time for list.insert(0, x) = 2.1753318309783936
Time for dequeue.appendleft(x) = 0.014408111572265625
```

Reply:
The `deque` starts to pay off after 100 hooks, given that at the end we would need to convert the `deque`s to lists in order to return a dictionary of the stages and their lists of hooks. I don't expect to have a test with so many hooks, but rather many tests with few hooks, which would mean that the `deque` solution might eventually be slower for this case.